src/rust/generators/sysmon-subgraph-generator/src/generator.rs

```diff
@@ -54,121 +49,76 @@ where
-        }
-
-        match (last_error, identities.identities.is_empty()) {
-            (Some(last_failure), true) => Err(Err(last_failure)),
-            (Some(last_failure), false) => Err(Ok((final_subgraph, last_failure))),
-            (None, _) => Ok(final_subgraph),
-        }
-    }
 }
 
 #[async_trait]
 impl<C> EventHandler for SysmonSubgraphGenerator<C>
 where
     C: Cache + Clone + Send + Sync + 'static,
 {
-    type InputEvent = Vec<u8>;
+    type InputEvent = Vec<Result<Event, crate::serialization::SysmonDecoderError>>;
     type OutputEvent = GraphDescription;
     type Error = SysmonGeneratorError;
 
     async fn handle_event(
         &mut self,
-        events: Vec<u8>,
+        events: Self::InputEvent,
         completed: &mut CompletedEvents,
     ) -> Result<Self::OutputEvent, Result<(Self::OutputEvent, Self::Error), Self::Error>> {
-        info!("Handling raw event");
-
-        /*
-           This iterator is taking a set of bytes of the logs, splitting the logs on newlines,
-           converting the byte sequences to utf-8 strings, and then filtering on the following criteria:
-               1. The line isn't empty
-               2. The line is not `\n` (to prevent issues with multiple newline sequences)
-               3. The line contains event with ID 1, 3, or 11
-
-           The event ids 1, 3, and 11 correspond to Process Creation, Network Connection, and File Creation
-           in that order.
-
-           https://docs.microsoft.com/en-us/sysinternals/downloads/sysmon#events
-        */
-        let events: Vec<_> = log_time!(
-            "event split",
-            events
-                .split(|i| &[*i][..] == &b"\n"[..])
-                .map(String::from_utf8_lossy)
-                .map(|s| s.to_string())
-                .filter(|event| {
-                    (!event.is_empty() && event != "\n")
-                        && (event.contains(&"EventID>1<"[..])
-                            || event.contains(&"EventID>3<"[..])
-                            || event.contains(&"EventID>11<"[..]))
-                })
-                .collect()
-        );
-
-        info!("Handling {} events", events.len());
-
-        let final_subgraph = self.process_events(events, completed).await;
+        info!("Processing {} incoming Sysmon events.", events.len());
+
+        let (deserialized_events, deserialization_errors): (Vec<_>, Vec<_>) =
+            events.into_iter().partition_map(|event| match event {
+                Ok(value) => Either::Left(value),
+                Err(error) => Either::Right(error),
+            });
+
+        for error in deserialization_errors {
+            error!("Deserialization error: {}", error);
+        }
+
+        let subgraph_results = deserialized_events
+            .into_iter()
+            .map(|event| GraphDescription::try_from(event));
+
+        // Report subgraph generation metrics
+        for result in subgraph_results.clone() {
```

Comment:
You can just call .iter() since yo're passing a reference in anyways - no need to clone.